packages/data-types/src/interfaces.ts

```diff
@@ -0,0 +1,95 @@
+
+import BaseType from './types/versions/base-type';
+
+export interface GraphQlResults {
+    results: string;
+    baseType: string;
+    customTypes: string[];
+}
+
+interface ObjectConfig {
+    [key: string]: any;
+}
+
+export interface MappingConfiguration {
+    typeName: string;
+    settings?: EsMapSettings;
+    mappingMetaData?: ObjectConfig;
+}
+
+export interface DataTypeManager {
+    toESMapping({}: MappingConfiguration): any;
+    toGraphQl(typeName?: string): string;
+    toGraphQLTypes(typeName?: string): GraphQlResults;
+    toXlucene(typeName: string|null|undefined, typeInjection?:string): XluceneMapping;
+}
+
+export type ElasticSearchTypes = 'long'|'integer'|'short'|'byte'|'double'|'float'|'keyword'|'text'|'boolean'|'ip'|'geo_point';
+
+export type AvailableTypes = 'Boolean'|'Date'|'Geo'|'IP'|'Byte'|'Double'|'Float'|'Integer'|'Keyword'|
+'Long'|'Short'|'Text'|'KeywordTokens'|'Hostname'|'KeywordCaseInsensitive'|'KeywordTokensCaseInsensitive'|'NgramTokens'|'Boundry';
+
+export type TypeConfig = {
+    type: AvailableTypes;
+};
+
+type ActualType = {
+    [key in AvailableTypes]: { new (field:string, config: TypeConfig): BaseType }
+};
+
+export interface DataTypeMapping {
+    [key: string]: ActualType;
+}
+
+export interface TypeConfigFields {
+    [key: string]: TypeConfig;
+}
+
+export type DataTypeConfig = {
+    fields: TypeConfigFields,
+    version: number;
+};
+
+export type EsTypeMapping = PropertyEsTypeMapping | BasicEsTypeMapping ;
+
+interface BasicEsTypeMapping {
+    type: ElasticSearchTypes;
+}
+
+interface PropertyEsTypeMapping {
+    type?: 'nested';
+    properties: {
+        [key: string]: BasicEsTypeMapping
+    };
+}
+
+export interface EsMapping {
```

Comment:
This should have proper naming:
```suggestion
export interface ESMapping {
```